homeassistant/components/cec.py

```diff
@@ -0,0 +1,120 @@
+"""
+CEC platform.
+
+Requires libcec + Python bindings.
+"""
+
+import logging
+import voluptuous as vol
+from homeassistant.const import EVENT_HOMEASSISTANT_START
+import homeassistant.helpers.config_validation as cv
+
+
+_LOGGER = logging.getLogger(__name__)
+_CEC = None
+DOMAIN = 'cec'
+SERVICE_SELECT_DEVICE = 'select_device'
+SERVICE_POWER_ON = 'power_on'
+SERVICE_STANDBY = 'standby'
+CONF_DEVICES = 'devices'
+ATTR_DEVICE = 'device'
+MAX_DEPTH = 4
+
+
+# pylint: disable=unnecessary-lambda
+DEVICE_SCHEMA = vol.Schema({
+    vol.All(cv.positive_int): vol.Any(lambda devices: DEVICE_SCHEMA(devices),
+                                      cv.string)
+})
+
+
+PLATFORM_SCHEMA = vol.Schema({
+    vol.Required(CONF_DEVICES): DEVICE_SCHEMA
+})
+
+
+def parse_mapping(mapping, parents=None):
+    """Parse configuration device mapping."""
+    if parents is None:
+        parents = []
+    for addr, val in mapping.items():
+        cur = parents + [str(addr)]
+        if isinstance(val, dict):
+            yield from parse_mapping(val, cur)
+        elif isinstance(val, str):
+            yield (val, cur)
+
+
+def pad_physical_address(addr):
+    """Right-pad a physical address."""
+    return addr + ['0'] * (MAX_DEPTH - len(addr))
+
+
+def setup(hass, config):
+    """Setup CEC capability."""
+    global _CEC
+
+    # cec is only available if libcec is properly installed
+    # and the Python bindings are accessible.
+    try:
+        import cec
```

Comment:
It works. Maybe I should change the filename to be safe? It could be `hdmi_cec.py`.